File: examples/companion-packages/ocaml-opam-app/lib/flags.ml
(* Flag definitions for the PawMatch app. Each flag is wired into at least
   one command's code path so the companion skills have realistic targets. *)

open Tinyflags

(* Flag name constants — string-literal sharing keeps callsites concise. *)
let home_check_followup = "home-check-followup"
let fee_breakdown_detailed = "fee-breakdown-detailed"
let long_stay_highlight = "long-stay-highlight"
let suggest_donate_after_adoption = "suggest-donate-after-adoption"
let show_charity_ratings = "show-charity-ratings"
let recommendation_strategy = "recommendation-strategy"
let match_quiz_depth = "match-quiz-depth"
let pet_card_style = "pet-card-style"
let donate_focus_default = "donate-focus-default"

let build_registry () : Tinyflags.t =
  make_exn
    [
      ( home_check_followup,
        Boolean (Bool.make_exn ~default:false ~rollout:25 ()) );
      (fee_breakdown_detailed, Boolean (Bool.make_exn ~default:true ()));
      (long_stay_highlight, Boolean (Bool.make_exn ~default:true ()));
      ( suggest_donate_after_adoption,
        Boolean (Bool.make_exn ~default:false ~rollout:50 ()) );
      (show_charity_ratings, Boolean (Bool.make_exn ~default:true ()));
      ( recommendation_strategy,
        VariantFlag
          (Variant.make_exn ~default:"match-quiz"
             ~rollout:[ ("longest-stay", 20) ]
             [ "popularity"; "match-quiz"; "longest-stay" ]) );
      ( match_quiz_depth,
        VariantFlag
          (Variant.make_exn ~default:"standard"
             [ "short"; "standard"; "thorough" ]) );
      ( pet_card_style,
        VariantFlag
          (Variant.make_exn ~default:"detailed"
             [ "compact"; "detailed"; "playful" ]) );
      ( donate_focus_default,
        VariantFlag
          (Variant.make_exn ~default:"all" [ "all"; "shelters"; "rescue" ])
      );
    ]

File: examples/companion-packages/lua-luarocks-app/src/pawmatch/flags.lua
--- PawMatch flag definitions wiring into the TinyFlags library.
--
-- Each flag is wired into at least one command's code path so the
-- companion skills (add-flag, rollout-review, cleanup-flag) have realistic
-- targets.

local tinyflags = require("tinyflags")

local M = {}

M.HOME_CHECK_FOLLOWUP            = "home-check-followup"
M.FEE_BREAKDOWN_DETAILED         = "fee-breakdown-detailed"
M.LONG_STAY_HIGHLIGHT            = "long-stay-highlight"
M.SUGGEST_DONATE_AFTER_ADOPTION  = "suggest-donate-after-adoption"
M.SHOW_CHARITY_RATINGS           = "show-charity-ratings"
M.RECOMMENDATION_STRATEGY        = "recommendation-strategy"
M.MATCH_QUIZ_DEPTH               = "match-quiz-depth"
M.PET_CARD_STYLE                 = "pet-card-style"
M.DONATE_FOCUS_DEFAULT           = "donate-focus-default"

function M.build_registry()
  return tinyflags.Registry({
    [M.HOME_CHECK_FOLLOWUP]            = tinyflags.BooleanFlag({ default = false, rollout = 25 }),
    [M.FEE_BREAKDOWN_DETAILED]         = tinyflags.BooleanFlag({ default = true }),
    [M.LONG_STAY_HIGHLIGHT]            = tinyflags.BooleanFlag({ default = true }),
    [M.SUGGEST_DONATE_AFTER_ADOPTION]  = tinyflags.BooleanFlag({ default = false, rollout = 50 }),
    [M.SHOW_CHARITY_RATINGS]           = tinyflags.BooleanFlag({ default = true }),
    [M.RECOMMENDATION_STRATEGY] = tinyflags.VariantFlag({
      variants = { "popularity", "match-quiz", "longest-stay" },
      default = "match-quiz",
      rollout = { ["longest-stay"] = 20 },
    }),
    [M.MATCH_QUIZ_DEPTH] = tinyflags.VariantFlag({
      variants = { "short", "standard", "thorough" },
      default = "standard",
    }),
    [M.PET_CARD_STYLE] = tinyflags.VariantFlag({
      variants = { "compact", "detailed", "playful" },
      default = "detailed",
    }),
    [M.DONATE_FOCUS_DEFAULT] = tinyflags.VariantFlag({
      variants = { "all", "shelters", "rescue" },
      default = "all",
    }),
  })
end

return M
